dglc/dglc_datamode_noevolve_mod.F90

```diff
@@ -380,25 +391,95 @@ subroutine dglc_datamode_noevolve_advance(pio_subsystem, io_type, io_format, &
       end if
 
       if (initialized_noevolve) then
+
          ! Compute Fgrg_rofi
          do ns = 1,num_icesheets
+
+            ! Get mesh info
             call ESMF_MeshGet(model_meshes(ns), elementdistGrid=distGrid, rc=rc)
             if (ChkErr(rc,__LINE__,u_FILE_u)) return
-            call ESMF_DistGridGet(distGrid, localDe=0, elementCount=lsize, rc=rc)
-            if (ChkErr(rc,__LINE__,u_FILE_u)) return
+            lsize = size(Fgrg_rofi(ns)%ptr)
```

Comment:
Now that you get lsize this way - which I think is an ideal way to get it - you can also get rid of the ESMF_MeshGet call: it looks like that was only there for the sake of the ESMF_DistGridGet call that you removed.

Reply:
Ok, this makes sense. Thanks for the comment explaining the code, too, that was really helpful!